code/modules/mob/living/carbon/human/species_types/oozelings.dm

```diff
@@ -182,6 +182,19 @@
 		H.gib()
 		return
 
+/datum/species/oozeling/help(mob/living/carbon/human/user, mob/living/carbon/human/target, datum/martial_art/attacker_style)
+	. = ..()
+	if(. && target != user && target.on_fire)
+		to_chat(user, "<span class='notice'>We begin to closely hug [target] to extinguish them...</span>")
+		target.visible_message("<span class='notice'>[user] begins to closely hug [target]...</span>", "<span class='boldnotice'>[user] holds you closely in a tight hug!</span>")
+		if(do_after(user, 1 SECONDS, target, IGNORE_HELD_ITEM))
+			target.visible_message("<span class='notice'>[user] extingushes [target] with a hug!</span>", "<span class='boldnotice'>[user] extingushes you with a hug!</span>", "<span class='italics'>You hear a fire sizzle out.</span>")
+			target.fire_stacks = max(target.fire_stacks - 5, 0)
+			if(target.fire_stacks <= 0)
+				target.ExtinguishMob()
+		else
+			target.visible_message("<span class='notice'>[target] wriggles out of [user]'s close hug!</span>", "<span class='notice'>You wriggle out of [user]'s close hug.</span>")
```

Comment:
Won't this message display even if the oozeling is the one that moves or otherwise disengages the hug? I feel like this should be worded differently to reflect that possibility

Reply:
Hm, how would you word it?

Reply:
I've been staring at this for ten minutes and not really had any good solution to this come up yet. 🤔
I might have to get back to you on this, I'm stumped right now.

Reply:
Yeah it's good enough and I can't think of anything better